Open Judge System/Services/OJS.Services/Business/ParticipantScores/Models/ParticipantScoreDataModel.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Expressions;
+using OJS.Data.Models;
+
+namespace OJS.Services.Business.ParticipantScores.Models
+{
+    public class ParticipantScoreDataModel
+    {
+        public Participant Participant { get; set; }
+
+        public bool IsOfficial { get; set; }
+
+        public string UserName { get; set; }
+
+        public int TotalScore { get; set; }
+
+        public static Expression<Func<Participant, ParticipantScoreDataModel>> FromParticipant =>
+            participant => new ParticipantScoreDataModel
+            {
+                Participant = participant,
+                IsOfficial = participant.IsOfficial,
+                UserName = participant.User.UserName,
+                TotalScore = participant.Scores
+                    .Where(ps => !ps.Problem.IsDeleted)
+                    .Select(ps => ps.Points)
+                    .DefaultIfEmpty(0)
+                    .Sum()
```

Comment:
Do you still need this?

Reply:
Yes, this is the totalScore of the participant without the current submission points. After that depending on number of things the points will be updated, but i need the value before update, in case that it needs to subtract the points for the current problem.

Reply:
This too I don't see it used anywhere

Reply:
Fixed